lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -74,6 +81,17 @@ def get(self, request, *args, **kwargs):
 
         course_tools = CourseToolsPluginManager.get_enabled_course_tools(request, course_key)
 
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        date_blocks = get_course_date_blocks(course, request.user, request, num_assignments=1)
+
+        # User locale settings
+        user_timezone_locale = user_timezone_locale_prefs(request)
+        user_timezone = user_timezone_locale['user_timezone']
+
+        dates_tab_link = request.build_absolute_uri(reverse('dates', args=[course.id]))
+        if course_home_mfe_dates_tab_is_active(course.id):
+            dates_tab_link = get_microfrontend_url(course_key=course.id, view_name='dates')
```

Comment:
We're going to be doing this pattern more and more yah? We should just have a utility method that encapsulates this for a given tab. All this code cares about is the right link, it shouldn't have to know about what to do based on whether an mfe tab waffle is active.

If you're feeling frisky, you can do that in this PR. Else one of us should do it next time we have to write code like this.